Genaral website update/overhaul

MagicLike opened the pull request:
- [x] Update links
  - [x] Add `rel="noopener"` to every link
  - [x] Update outdated links
  - [x] Remove trailing slashes
  - [x] Remove dead links
- [x] Spelling
- [x] Code formatting
- [x] Update FAQ
- [x] Remove inline CSS
- [x] Comply with HTML standards
- [x] Add donation buttons/widgets
- [x] Rearrange devices according to README
- [x] Adding border-radius to menu to fit website theme (#18)
- [x] Use transition: ease-in on hover for certain areas like the menu, the Download buttons, etc. (#18)
- [x] #13
- [x] #12
- [x] Document website build workflow in README
  - [x] Create a GitHub Actions build workflow (see in action [here](https://github.com/Magic-Tests/openandroidinstaller-dev.github.io/actions))
- [x] Add 404 page

MagicLike committed:
Updated docs links

MagicLike committed:
Updated `unlock-bootloader`

MagicLike committed:
Formatting

MagicLike committed:
Updated TOC to comply with HTML standards

MagicLike committed:
Updated links

- Added `rel="noopener"` to every link
- Updated outdated links
- Removed trailing slashes

MagicLike committed:
Spelling & formatting

MagicLike committed:
Added help links

MagicLike committed:
Move CSS from `b871dea` to `custom.css`

MagicLike committed:
Updated more links

- Added `rel="noopener"` to every link
- Updated outdated links
- Removed trailing slashes
- Removed dead links

MagicLike committed:
Removed inline CSS

MagicLike committed:
Formatting

MagicLike committed:
Added button `type` to `Submit` button in feedback

MagicLike committed:
Rearranged devices according to README

MagicLike committed:
Spelling

MagicLike committed:
Added new donation buttons

These buttons replace the old two links and add Liberapay.

tsterbak commented on templates/download.html:
```diff
@@ -114,7 +115,7 @@ <h2 class="has-text-white mt-2 mb-3 is-size-1 is-size-3-mobile has-text-weight-b
         <div class="mx-auto box p-6 has-background-light has-text-centered">
           <a class="button is-link is-fullwidth" id="contribute-button"
-            href="mailto:help@openandroidinstaller.org">Write a mail</a>
+            href="mailto:help@openandroidinstaller.org">Write a email</a>
```
```suggestion
            href="mailto:help@openandroidinstaller.org">Write an Email</a>
```

MagicLike replied:
fixed in 431990f

tsterbak committed:
Add public files

MagicLike committed:
Corrected spelling for e-mail

MagicLike committed:
Rounded up!

Added border radius to more website elements to fit theme

MagicLike committed:
Added on hover ease for buttons and tiles

MagicLike committed:
Deleted unused and outdated images

MagicLike committed:
Closed uncloed `div`s

reolves #13

MagicLike committed:
Added 404 page

MagicLike committed:
Spelling

MagicLike committed:
Corrected section descriptions

MagicLike committed:
Added .htaccess

- CSP
- STS
- X-Content
- RP
- HTTP Cache
- Removing file extensions from links ✨

resolves #12

MagicLike committed:
Use the right e-mail address for the right use case

MagicLike committed:
Removed unused sponsor image

Newer version is in use.

MagicLike committed:
Formatting

MagicLike committed:
Removed domain from internal links

MagicLike committed:
Added `Donation options` section title to `custom.css`

MagicLike committed:
Merge remote-tracking branch 'origin/page-updates' into docs-update

MagicLike committed:
Merge pull request #21 from openandroidinstaller-dev/docs-update

Docs update

MagicLike committed:
Revert "Add public files"

This reverts commit 10b00659b4a83f53f11ebf6889faaf57bb17865c.

MagicLike committed:
Documented website build workflow

MagicLike committed:
Bump workflow dependencies

MagicLike committed:
Implement automatic website builds

MagicLike commented:
The only thing missing is https://github.com/openandroidinstaller-dev/openandroidinstaller-dev.github.io/issues/16#issuecomment-1729855472

@tsterbak

tsterbak commented:
Are you sure the new workflow works?

Also, it would be great if the build and deployment would be triggered by adding commits to the main branch, and not only through PRs.

tsterbak commented on templates/privacy.html:
```diff
@@ -160,8 +161,8 @@ <h2 class="has-text-white mt-2 mb-3 is-size-1 is-size-3-mobile has-text-weight-b
-          <a class="button is-link is-fullwidth" id="contribute-button"
-            href="mailto:help@openandroidinstaller.org">Write a mail</a>
+          <a class="button is-link is-fullwidth hover-transition" id="contribute-button"
+            href="mailto:hello@openandroidinstaller.org">Write an e-mail</a>
```
There is a reason why this is `help@`. I want to know where people came from :)

MagicLike replied:
Ooops O.o

MagicLike replied:
fixed now

tsterbak replied:
No worries, It was written nowhere :) So you couldn't know

MagicLike committed:
Build can now be triggered via commit

MagicLike commented:
> Also, it would be great if the build and deployment would be triggered by adding commits to the main branch, and not only through PRs.

fixed!

MagicLike committed:
Corrected e-mail addresses again

MagicLike commented:
> Are you sure the new workflow works?

I tested it multiple times, it should work...

tsterbak reviewed (approved):
LGTM :)

tsterbak commented:
I'm planning to push the next release of the Installer on Friday (next week) and then publish this PR.

Thanks a lot for cleaning and improving the website! :blush:

MagicLike commented:
> I'm planning to push the next release of the Installer on Friday (next week) and then publish this PR.
> 
> Thanks a lot for cleaning and improving the website! 😊

No problem! :D - But there is still stuff to do after that ^^

tsterbak commented:
Did you test the new github workflow? Because every time I touch this stuff something breaks if I don't test it.
Otherwise, I will remove the workflow update from this PR and try to get it in after the release.

MagicLike commented:
> Did you test the new github workflow? Because every time I touch this stuff something breaks if I don't test it. Otherwise, I will remove the workflow update from this PR and try to get it in after the release.

I did - multiple times, One example can be seen here: https://github.com/Magic-Tests/openandroidinstaller-dev.github.io/actions

But if you feel uncomfortable about it, feel free to not include it for now!

tsterbak commented:
> I did - multiple times, One example can be seen here: https://github.com/Magic-Tests/openandroidinstaller-dev.github.io/actions

Okay great, great! 😊 looks perfect 👏 thank you for clarifying.

> But if you feel uncomfortable about it, feel free to not include it for now!

No, I want it in and it looks solid 🙂

tsterbak committed:
Bump version and device count

tsterbak committed:
Add new devices

tsterbak commented:
The build workflow didn't work properly, since it has no permission to push to main. I guess this is a general issue and cannot be avoided if the main branch is protected and only allows commits through PRs... Maybe pushing to the deployment branch might work? I'm not sure

MagicLike commented:
> The build workflow didn't work properly, since it has no permission to push to main. I guess this is a general issue and cannot be avoided if the main branch is protected and only allows commits through PRs... Maybe pushing to the deployment branch might work? I'm not sure

I'm gonna do some research later 🤔

tsterbak commented:
> I'm gonna do some research later 🤔

Seems to be an annoying issue :D

But other than that it would have worked great! And it also looks great now, thanks a lot :)

MagicLike commented:
> > I'm gonna do some research later 🤔
> 
> Seems to be an annoying issue :D
> 
> But other than that it would have worked great! And it also looks great now, thanks a lot :)

Hmmm it should be possible for you to allow the Actions user to commit directly to main in the branch protection section :thinking: (Should be `github-actions[bot]`)

tsterbak commented:
> Hmmm it should be possible for you to allow the Actions user to commit directly to main in the branch protection section 🤔 (Should be github-actions[bot])

Where did you get that info from? My understanding was, that has been removed, since it technically allows to bypass branch protection for anyone who can create branches on the repo.

MagicLike commented:
> > Hmmm it should be possible for you to allow the Actions user to commit directly to main in the branch protection section 🤔 (Should be github-actions[bot])
> 
> Where did you get that info from? My understanding was, that has been removed, since it technically allows to bypass branch protection for anyone who can create branches on the repo.

I made a quick search and found this: https://overflow.adminforge.de/questions/69263843/how-to-push-to-protected-main-branches-in-a-github-action#72515781

tsterbak commented:
> I made a quick search and found this: https://overflow.adminforge.de/questions/69263843/how-to-push-to-protected-main-branches-in-a-github-action#72515781

Going forward, this will allow anyone allowed to create branches on the repo to basically commit with admin privileges. And some comments say it doesn't even work.

I guess I rather prefer to just have html commited in the public folder on main deployed? Seems very easy and I build the files locally first either way. :thinking:  I will think about it

MagicLike commented:
> > I made a quick search and found this: https://overflow.adminforge.de/questions/69263843/how-to-push-to-protected-main-branches-in-a-github-action#72515781
> 
> Going forward, this will allow anyone allowed to create branches on the repo to basically commit with admin privileges. And some comments say it doesn't even work.
> 
> I guess I rather prefer to just have html commited in the public folder on main deployed? Seems very easy and I build the files locally first either way. 🤔 I will think about it

wait, there should also be an option to let GH Actions run alongside a PR, for example, when it gets approved it will build the website on the PR branch and then it can get merged...

tsterbak commented:
> wait, there should also be an option to let GH Actions run alongside a PR, for example, when it gets approved it will build the website on the PR branch and then it can get merged...

That would be great I guess :)
Could we even get a preview deployment of the website in this way?

MagicLike commented:
> > wait, there should also be an option to let GH Actions run alongside a PR, for example, when it gets approved it will build the website on the PR branch and then it can get merged...
> 
> That would be great I guess :)
> Could we even get a preview deployment of the website in this way?

That should also be possible... - But I will need to investigate that further in the next few days (I have never done that before)
